sdk/core/azure-core/azure/core/tracing/_inference_api_injector.py

```diff
@@ -0,0 +1,370 @@
+    if item.index:
+        accumulate["index"] = item.index
+    if item.delta.content:
+        accumulate.setdefault("message", {})
+        accumulate["message"].setdefault("content", "")
+        accumulate["message"]["content"] += item.delta.content
+    if item.delta.tool_calls:
+        accumulate.setdefault("message", {})
+        accumulate["message"].setdefault("tool_calls", [])
+        for tool_call in item.delta.tool_calls:
+            if tool_call.id:
+                accumulate["message"]["tool_calls"].append({"id": tool_call.id, "type": "", "function": {"name": "", "arguments": ""}})
+            if tool_call.type:
+                accumulate["message"]["tool_calls"][-1]["type"] = tool_call.type
+            if tool_call.function and tool_call.function.name:
+                accumulate["message"]["tool_calls"][-1]["function"]["name"] = tool_call.function.name
+            if tool_call.function and tool_call.function.arguments:
+                accumulate["message"]["tool_calls"][-1]["function"]["arguments"] += tool_call.function.arguments
+
+
+def _wrapped_stream(stream_obj: _models.StreamingChatCompletions, span: AbstractSpan) ->  _models.StreamingChatCompletions:
+    class StreamWrapper(_models.StreamingChatCompletions):
+        def __init__(self, stream_obj):
+            super().__init__(stream_obj._response)
+
+        def __iter__(self) -> Iterator[_models.StreamingChatCompletionsUpdate]:
+            global _trace_inference_content
+            try:
+                accumulate: dict[str, Any] = {}
+                for chunk in stream_obj:
+                    for item in chunk.choices:
+                        _accumulate_response(item, accumulate)
+                    yield chunk
+
+                if _trace_inference_content:
+                    span.span_instance.add_event(
+                        name="gen_ai.choice",
+                        attributes={
+                            "get_ai.system": "openai",
+                            "gen_ai.event.content": json.dumps(accumulate)
+                        }
+                    )
+                _add_response_chat_attributes(span, chunk)
+
+            except Exception as exc:
+                _set_attributes(span, ("error.type", exc.__class__.__name__))
+                raise
+
+            finally:
+                if stream_obj._done is False:
+                    if accumulate.get("finish_reason") is None:
+                        accumulate["finish_reason"] = "error"
+                    if _trace_inference_content:
+                        span.span_instance.add_event(
+                            name="gen_ai.choice",
+                            attributes={
+                                "get_ai.system": "openai",
+                                "gen_ai.event.content": json.dumps(accumulate)
+                            }
+                        )
+                span.finish()
+
+    return StreamWrapper(stream_obj)
+
```

Comment:
I need to talk to you about streaming response to understand what the logging requirements are.  Note that you will also need to handle streaming of a model response that includes a tool call. The behavior changes between models (e.g. Mistral Large and gpt-4o). We can talk more over phone.